packages/fluentui/react-northstar/src/components/Tree/hooks/flattenTree.ts

```diff
@@ -129,20 +130,21 @@ function flattenSubTree(
         selectedItemIds,
       );
 
-      if (selectedChildrenNum === selectableChildrenNum) {
-        flatTree[id].selected = true;
-        selectedNum++;
-      } else if (selectedChildrenNum > 0) {
-        flatTree[id].selected = 'indeterminate';
-        selectedNum += 0.5; // trick to propagate indeterminate state to ancestors
+      if (selectable) {
+        if (selectedChildrenNum === selectableChildrenNum) {
+          flatTree[id].selected = true;
+          selectedNum++;
+        } else if (selectedChildrenNum > 0) {
+          flatTree[id].selected = 'indeterminate';
+          selectedNum += 0.5; // trick to propagate indeterminate state to ancestors
+        }
       }
-    } else if (selectedItemIds.indexOf(id) >= 0) {
+    } else if (selectable && selectedItemIds.indexOf(id) >= 0) {
       flatTree[id].selected = true;
       selectedNum++;
     }
 
-    if (selectable !== false) {
-      // by default item is selectable, unless selectable=false specified
+    if (selectable) {
```

Comment:
```suggestion
    if (selectable) {
        if(selectedItemIds.indexOf(id) >= 0) {        
                flatTree[id].selected = true; 
         }
```

Reply:
Hmmm I get what you mean, putting all the selection logic together.
But I prefer to leave the `if(selectedItemIds.indexOf(id) >= 0)` block with the previous `if(hasSubtree)...else...`, because this is a condition to check for selection state of leaf node.

Reply:
For me it's kind of repeating logic, but up to you. @ling1726 WDYT?

Reply:
As we discussed offline, this should be correct because there still needs to be logic to handle leaf nodes. Hoisting it to under `hasSubtree` will not solve that case

As far as not duplicating logic goes, might be useful to create helper function for that `setNodeSelected(flatTree[id], SELECTED/INDETERMINATE)`

Reply:
As we discussed offline @YuanboXue-Amber just remove the `if(hasSubtree)` block